src/Core/OrganizationFeatures/OrganizationSubscriptions/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -79,6 +83,25 @@ public async Task AdjustServiceAccountsAsync(Organization organization, int smSe
         await UpdateSubscriptionAsync(secretsManagerSubscriptionUpdate);
     }
 
+    public async Task AutoAddSmSeatsAsync(Organization organization, int smSeatAdjustment)
+    {
+        if (_globalSettings.SelfHosted)
+        {
+            throw new BadRequestException("Cannot autoscale on a self-hosted instance.");
+        }
+
+        var update = new SecretsManagerSubscriptionUpdate(
+            organization, seatAdjustment: smSeatAdjustment, maxAutoscaleSeats: organization?.MaxAutoscaleSmSeats,
+            serviceAccountAdjustment: 0, maxAutoscaleServiceAccounts: organization?.MaxAutoscaleSmServiceAccounts);
+
+        if (organization.MaxAutoscaleSmSeats.HasValue && update.SmSeats > organization.MaxAutoscaleSmSeats.Value)
+        {
+            throw new BadRequestException("Secrets Manager seat limit has been reached.");
+        }
```

Comment:
The main update command will throw an exception in this case anyway, `Cannot set max seat autoscaling below seat count.` But this is a more appropriate error message for the user to see when trying to autoscale.